src/lib/ui/AccessCodeTable/AccessCodeTable.tsx

```diff
@@ -59,52 +63,70 @@ export function AccessCodeTable(
         </TableTitle>
       </TableHeader>
       <TableBody>
-        {accessCodes.map((code) => (
-          <TableRow
-            key={code.access_code_id}
-            onClick={() => {
-              selectAccessCode(code)
-            }}
-          >
-            <TableCell className='seam-icon-cell'>
-              <div>
-                <AccessCodeKeyIcon />
-              </div>
-            </TableCell>
-            <TableCell className='seam-name-cell'>
-              <Title>{code.name}</Title>
-              <CodeDetails accessCode={code} />
-            </TableCell>
-            <TableCell className='seam-action-cell'>
-              <MoreActionsMenu
-                MenuProps={{
-                  BackgroundProps: {
-                    className: 'seam-access-code-table-action-menu',
-                  },
-                }}
-              >
-                <MenuItem
-                  onClick={() => {
-                    void copy(code.code ?? '')
-                  }}
-                >
-                  <div className='menu-item-copy'>
-                    <span>
-                      {t.copyCode} - {code.code}
-                    </span>
-                    <CopyIcon />
-                  </div>
-                </MenuItem>
-              </MoreActionsMenu>
-            </TableCell>
-          </TableRow>
-        ))}
+        <Body accessCodes={accessCodes} selectAccessCode={selectAccessCode} />
       </TableBody>
     </div>
   )
 }
 
+function Body(props: {
+  accessCodes: Array<UseAccessCodesData[number]>
+  selectAccessCode: (accessCode: AccessCode) => void
+}) {
+  const { accessCodes, selectAccessCode } = props
+
+  if (accessCodes.length === 0) {
+    return <EmptyPlaceholder>{t.noAccessCodesMessage}</EmptyPlaceholder>
+  }
+
+  return (
+    <>
+      {accessCodes.map((code) => (
```

Comment:
Regarding nesting, can we go one step further and keep map n functions to a single component, e.g., like we do here: https://github.com/seamapi/react/blob/7a2b7e4bbe672b464c28754ec39cef021d424161/src/lib/ui/DeviceTable/DeviceTable.tsx#L72

Reply:
Sure, updated in https://github.com/seamapi/react/pull/172/commits/ba74c6ee34a621c898fd8141be2640cf46a9a373